### PR TITLE
Crap ton of breaking api changes and implementation

### PR DESCRIPTION
**Pull request by lucyy-mc**
See commit log for details

### CODE REVIEW

**Comment by lucyy-mc** [comment]
Fixes #10 by replacing with github issues for things that dont need explicit todos

File: checkstyle.xml [hunk 1]
```diff
@@ -1,348 +0,0 @@
-<?xml version="1.0"?>
```
Comment (RedaAM) [inline checkstyle.xml:1]: hehe its gone :D

File: watercore-api/src/main/java/me/lucyy/watercore/api/module/ModuleManager.java [hunk 2]
```diff
@@ -70,13 +79,23 @@
...
 	 */
 	void unloadModule(WaterModule module);
+
+	/**
+	 * Reloads all modules. This will unregister all commands and events, dispose of the module instance
+	 * and then create a new one.
+	 *
+	 * @since 1.0.0
+	 */
+	void reloadModules();
```
Comment (RedaAM) [inline watercore-api/src/main/java/me/lucyy/watercore/api/module/ModuleManager.java:100]: O.O Module reloads, That's hella handy. ngl.

File: watercore-api/src/main/java/me/lucyy/watercore/api/version/SemanticVersion.java [hunk 3]
```diff
@@ -63,20 +71,25 @@ public int getPatch() {
 	 * including a hyphen in theversion.
```
Comment (lucyy-mc) [inline watercore-api/src/main/java/me/lucyy/watercore/api/version/SemanticVersion.java:71]: is that a typo i see :eyes:

Comment (RedaAM) [reply watercore-api/src/main/java/me/lucyy/watercore/api/version/SemanticVersion.java:71]: Not as bad as the spigot potioneffecttype docs xD
(see pic attached)
Capture d’écran 2021-05-03 à 20 18 55

**Review by RedaAM — COMMENTED** [review]
its good idk what to say lmao, it is what it is 👍 :D
here's a water for your hard work :D
AAAAwaterwater

**Review by RedaAM — APPROVED** [review]

**Pull request merged**